pkg/transport/pool.go

```diff
@@ -80,16 +139,37 @@ func (tp *TransportPool) newClient(name string) (*Client, error) {
 		return nil, err
 	}
 
+	headers := map[string]string{}
+	headerKeys := []string{}
```

Comment:
Could pre-allocate the exact size for both

Reply:
- [ ] this